server/service/microsoft_mdm.go

```diff
@@ -1265,6 +1266,96 @@ func (svc *Service) isFleetdPresentOnDevice(ctx context.Context, deviceID string
 	return true, nil
 }
 
+func (svc *Service) enqueueInstallFleetdCommand(ctx context.Context, deviceID string) error {
+	secrets, err := svc.ds.GetEnrollSecrets(ctx, nil)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "getting enroll secrets")
+	}
+
+	if len(secrets) == 0 {
+		level.Warn(svc.logger).Log("msg", "unable to find a global enroll secret to install fleetd")
+		return nil
+	}
+
+	appCfg, err := svc.ds.AppConfig(ctx)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "getting app config")
+	}
+	fleetURL := appCfg.ServerSettings.ServerURL
+	globalEnrollSecret := secrets[0].Secret
+	addCommandUUID := uuid.NewString()
+	execCommandUUID := uuid.NewString()
+
+	rawAddCmd := []byte(`
+<Add>
+	<CmdID>` + addCommandUUID + `</CmdID>
+	<Item>
+		<Target>
+			<LocURI>./Device/Vendor/MSFT/EnterpriseDesktopAppManagement/MSI/%7BA427C0AA-E2D5-40DF-ACE8-0D726A6BE096%7D/DownloadInstall</LocURI>
```

Comment:
Seems like this is our Fleet GUID? Maybe we should extract it as a constant so it can also be reused for tests instead of copying it every time.

Reply:
great idea, why didn't I think of this while struggling to copy and paste that long string is beyond me. Thanks!